src/lib/protocols/tls.c

```diff
@@ -977,7 +993,8 @@ static int ndpi_search_tls_tcp(struct ndpi_detection_module_struct *ndpi_struct,
      generic "extra function" code doesn't honour protocol bitmask.
      TODO: handle that in ndpi_main.c for all the protocols */
   if(packet->payload_packet_len == 0 ||
-     packet->tcp_retransmission) {
+     packet->tcp_retransmission ||
+     tcp_ack_padding(packet)) {
```

Comment:
Does the TCP ACK padding only occur for TLS?

Reply:
In the process of being clarified.

Reply:
Statistics showed that padding does not depend on the application protocol.